File: datamodel/src/main/java/datamodel/clinical/LabValue.java
package datamodel.clinical;

import java.time.LocalDate;

import org.immutables.value.Value;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

@Value.Immutable
@Value.Style(passAnnotations = { NotNull.class, Nullable.class })
public abstract class LabValue {

    @NotNull
    public abstract LocalDate date();

    @NotNull
    public abstract String code();

    @NotNull
    public abstract String name();

    @NotNull
    public abstract String comparator();

    public abstract double value();

    @NotNull
    public abstract LabUnit unit();

    @Nullable
    public abstract Double refLimitLow();

    @Nullable
    public abstract Double refLimitUp();

    @Nullable
    public abstract Boolean isOutsideRef();

}

File: algo/src/main/kotlin/com/hartwig/actin/datamodel/clinical/LabValue.kt
package com.hartwig.actin.datamodel.clinical

import java.time.LocalDate

data class LabValue(
    val date: LocalDate,
    val measurement: LabMeasurement,
    val comparator: String,
    val value: Double,
    val unit: LabUnit,
    val refLimitLow: Double? = null,
    val refLimitUp: Double? = null,
    val isOutsideRef: Boolean? = null
)
